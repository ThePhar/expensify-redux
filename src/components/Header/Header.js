import React           from 'react';
import { connect }     from 'react-redux';
import { Link }        from 'react-router-dom';
import { startLogout } from '../../actions/auth';

import './Header.scss';

export const Header = ({ startLogout }) => (
    <header className="header">
        <Link className="header__title" to="/dashboard">
            <h1>Expensify</h1>
        </Link>
        <button className="header__button" onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);