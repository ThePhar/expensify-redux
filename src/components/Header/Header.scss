@import '../../index.scss';

.header {
    display: flex;
    background-color: #364051;
    justify-content: space-between;
    padding: 0 $l-size;

    .header__title {
        color: white;
        text-decoration: none;
        margin: 0;
        line-height: 1;
        padding: 0;
    }

    .header__button {
        background: none;
        border: none;
        color: white;
        font-size: $l-size;
        font-weight: 300;
        line-height: 1;
        padding: 0;
    }
}