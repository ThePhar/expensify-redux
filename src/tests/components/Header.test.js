import { shallow } from 'enzyme';
import React       from 'react';
import { Header }  from '../../components/Header/Header';

let startLogout, wrapper;
beforeEach(() => {
    startLogout = jest.fn();
    wrapper = shallow(<Header startLogout={startLogout} />)
});

test('Should render Header correctly.', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout when clicked.', () => {
     wrapper.find('button').simulate('click');
     expect(startLogout).toHaveBeenCalled();
});