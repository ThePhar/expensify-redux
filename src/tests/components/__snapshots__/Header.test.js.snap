// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly. 1`] = `
<header
  className="header"
>
  <Link
    className="header__title"
    to="/dashboard"
  >
    <h1>
      Expensify
    </h1>
  </Link>
  <button
    className="header__button"
    onClick={[MockFunction]}
  >
    Logout
  </button>
</header>
`;
